api/envoy/config/route/v3/route_components.proto

```diff
@@ -437,6 +437,13 @@ message WeightedCluster {
   // configuration file will be used as the default weight. See the :ref:`runtime documentation
   // <operations_runtime>` for how key names map to the underlying implementation.
   string runtime_key_prefix = 2;
+
+  oneof random_value_specifier {
+    // Specifies the header name that is used to look up the random value passed in the request header.
+    // This is used to ensure consistent cluster picking across multiple proxy levels for weighted traffic.
+    // Signal-valued header is expected here as we only want to have one value throughout the process for the consistency.
```

Comment:
Typo 'Signal' -> "Single"
Also add that the value is expected to be a valid unsigned number between 1 and 2^64-1. Also add that if the header is not present or invalid, then Envoy fall back to using internally generated random number.

Reply:
Yep. Done.